File: 卡码/KamaCoder/Python/test12.py
#12. 打印数字图形
import sys

def printSpace(n):
	for i in range(n):
		print(" ",end="")

def printLine(n):
	for i in range(1,n+1):
		print(i,end="")
	for i in range(n-1,0,-1):
		print(i,end="")
	print()

def main():
	for line in sys.stdin:
		n=int(line)
		for i in range(1,n+1):
			printSpace(n-i)
			printLine(i)
		for i in range(n-1,0,-1):
			printSpace(n-i)
			printLine(i)


if __name__ == '__main__':
	main()

File: 卡码/KamaCoder/C++/test12.cpp
//12. 打印数字图形
#include <iostream>
using namespace std;

void printSpace(int n){
	for(int i=0;i<n;i++) printf(" ");
}

void printLine(int n){
	for(int i=1;i<=n;i++) printf("%d",i);
	for(int i=n-1;i>=1;i--) printf("%d",i);
	printf("\n");
}

int main(){
	int n;
	while(scanf("%d",&n)!=EOF){
		for(int i=1;i<=n;i++){
			printSpace(n-i);
			printLine(i);
		}
		for(int i=n-1;i>=1;i--){
			printSpace(n-i);
			printLine(i);
		}
	}
	return 0;
}